int32 id
duration duration
